### ISSUE TITLE
Feature request: backup/export with QR codes

### ISSUE THREAD

**Issue by anastrophe** [jamie-mh/AuthenticatorPro#51]
Having searched off and on for some time now for an alternative to Authenticator+, a terrific 2FA app, but pretty much abandonware at this point (last update Dec 2018 and no responses to inquires), I finally came upon AuthenticatorPro today, having tried several others at the same time - Aegis, 2FA authenticator, Microsoft's stab at it etc. - none of which made the cut.

It's darned near finally a perfect replacement for me. Except for one thing - and that one thing that's lacking is precisely what made it easy to switch from Auth+ to AuthPro: backup/export that includes an html output file with QR codes for each of the credentials, see attached:

<img width="752" alt="2020-09-14 19_09_24-Authenticator Plus" src="https://user-images.githubusercontent.com/1012545/93157282-c48cc380-f6be-11ea-9359-e7c4b4df1fc2.png">

Any chance of that being implemented? It makes migration a breeze.

Thanks in advance!

**Comment by jamie-mh** [comment]
Hi @anastrophe ,

Unfortunately, this format is quite unfriendly to users, unless you are very technically inclined, it could be confusing. I can envisage people mistaking these html backups with regular backup files and attempting to restore them.

However, this might tie into an idea that I've had recently. I'm thinking of creating a web-app than can read and display Authenticator Pro backup files offline inside the browser. This will help users decrypt their files without using OpenSSL and the command line. QR codes can be a format that the web-app displays.

Is this something that makes sense to you and would be interested in?

**Comment by anastrophe** [comment]
Hmm. I would have to disagree that it's unfriendly to users. Perhaps I didn't share enough info - it's a separate option from regular backup/restore functions, identified as 

> Export to other apps - 
> Export as Text:
> Export accounts as text and html file in an encrypted zip

The idea being that it's for that generally rare usage when one is either migrating to a new phone, or migrating to new 2fa software. Being able just load the file into one's browser - strictly locally, of course - then just snap images of each QR code, makes that migration extremely simple - and far easier.

What you describe in the second paragraph is pretty much exactly what this function is - a way to replicate all of the original QR code images that were used when initially creating a new 2fa credential - without having to iterate your way through a hundred different websites to re-register.

**Comment by jamie-mh** [comment]
I believe I understand exactly what you're saying.
You're suggesting that when the user clicks on the "Backup" menu option, a second menu appears with a list of choices. In effect, this greatly increases the complexity of the backup option as it's unclear which options can be restored within the app itself.
I understand the merit of what you're suggesting, but in my opinion this should be a separate tool. As I suggested before, a web-app that reads the regular backup files offline in the browser and creates a PDF with QR codes or something similar.

At the moment, I receive countless emails from people who are confused about various functions of the app. I guarantee that adding an extra choice in the backup flow will absolutely cause confusion :)

**Comment by anastrophe** [comment]
Thanks Jamie. I certainly understand the demands of maintaining open source software, for sure. I'd recommend if you have a moment that you just download the Authenticator Plus application and have a look at how it's implemented there - it's pretty straightforward. However, I'm a well-versed techie myself, systems-administrator, so digging into options and tweaking stuff is what i do for a living! Perhaps not your typical user.
I agree that a mechanism as you describe is virtually the same as what I'm suggesting. Cheers!

**Comment by bryanyork** [comment]
Just another +1 to a solution here. Would rather not re-setup my 2FA codes from Authenticator+

Web app should be fine, but another hidden option within advanced settings which wouldn't confuse users might be good as well

**Comment by jamie-mh** [comment]
Hi,

After looking into the idea of a web-app, I found the performance to be rather abysmal (30 seconds to generate the QR codes in the browser). So I've implemented the feature in the app, with a detailed description of each backup type in order to prevent confusion. In hindsight, generating the file within the app is much more convenient. It can be saved to the device or cloud storage just like the regular backup files.

The feature will be available in the next release.
